projects/benqi-ignite/functions/registerWithErc20Fee.ts

```diff
@@ -0,0 +1,167 @@
+    }
+
+    const { account, chainId } = wallet.data;
+
+    const register = parseRegister(props);
+
+    if (!register.success) {
+        return toResult(register.errorMessage, true);
+    }
+
+    const paymentMethodAddress = ERC20_PAYMENT_METHODS[props.paymentMethod];
+
+    if (!paymentMethodAddress) {
+        return toResult(`Payment method ${props.paymentMethod} is not supported`, true);
+    }
+
+    const transactions: TransactionParams[] = [];
+    const provider = getProvider(chainId);
+    const time = VALIDATION_DURATION_TIME[register.data.validationDuration];
+
+    await notify('Started estimating fee for registerWithErc20Fee transaction...');
+
+    await notify('Fetching price feed contracts addresses...');
+
+    const [paymentMethodPriceFeed, avaxPriceFeed] = await provider.multicall({
+        contracts: [
+            {
+                address: IGNITE_ADDRESS,
+                abi: igniteAbi,
+                functionName: 'priceFeeds',
+                args: [paymentMethodAddress],
+            },
+            {
+                address: IGNITE_ADDRESS,
+                abi: igniteAbi,
+                functionName: 'priceFeeds',
+                args: [AVAX_ADDRESS],
+            },
+        ],
+    });
+
+    if (paymentMethodPriceFeed.status !== 'success') return toResult(paymentMethodPriceFeed.error.message, true);
+
+    if (avaxPriceFeed.status !== 'success') return toResult(avaxPriceFeed.error.message, true);
+
+    await notify('Fetching latest prices from feeds...');
+
+    const [paymentMethodLatestRound, avaxLatestRound, paymentMethodDecimals] = await provider.multicall({
+        contracts: [
+            {
+                address: paymentMethodPriceFeed.result,
+                abi: eacAggregatorProxyAbi,
+                functionName: 'latestRoundData',
+                args: [],
+            },
+            {
+                address: avaxPriceFeed.result,
+                abi: eacAggregatorProxyAbi,
+                functionName: 'latestRoundData',
+                args: [],
+            },
+            {
+                address: paymentMethodAddress,
+                abi: erc20Abi,
+                functionName: 'decimals',
+                args: [],
+            },
+        ],
+    });
+
+    if (paymentMethodLatestRound.status !== 'success') return toResult(paymentMethodLatestRound.error.message, true);
+
+    if (avaxLatestRound.status !== 'success') return toResult(avaxLatestRound.error.message, true);
+
+    if (paymentMethodDecimals.status !== 'success') return toResult(paymentMethodDecimals.error.message, true);
+
+    await notify('Calculating fee...');
+
+    const feeInAvax = AVAX_REGISTRATION_FEE[register.data.validationDuration];
+    const avaxPrice = avaxLatestRound.result[1];
+    const tokenPrice = paymentMethodLatestRound.result[1];
+
+    let tokenAmount = (avaxPrice * feeInAvax) / tokenPrice / 10n ** BigInt(AVAX_DECIMALS - paymentMethodDecimals.result);
```

Comment:
Would be helpful to give some description / comment / example for following calculation

Reply:
I thought the names of variables make this kinda self explanatory. Honestly I'm not convinced that adding comment like:
```ts
// Token amount is 1/10 of `(avaxPrice * feeInAvax) / tokenPrice`
```
would make it more readable 🤷

Reply:
Example comment:
```
// Computation taken from: Ignite.sol(0x2575472517d23Ae4B9D4670F84F0c9c4eFC3876c).registerWithErc20Fee#L293
```

Reply:
https://github.com/RealWagmi/anon-integration-guide/pull/16/commits/df35f7d55aa7e841090c9f887157b71d3b98270d